Fix Color Picker corrupting RAW mode colors #21860

chaosdrop opened the pull request:
This allows raw mode colors to be set and not corrupted later.

Fix #21860

aaronfranke commented on scene/gui/color_picker.cpp:
```diff
@@ -76,8 +76,9 @@ void ColorPicker::_notification(int p_what) {
 void ColorPicker::set_focus_on_line_edit() {
-
-	c_text->call_deferred("grab_focus");
+	if (raw_mode_enabled == false) {
```
Should be `if (!raw_mode_enabled) {`

aaronfranke commented on scene/gui/color_picker.cpp:
```diff
@@ -507,8 +508,9 @@ void ColorPicker::_focus_exit() {
-	_focus_exit();
+	//Commented out as a temporary fix for issue #21860. So that raw mode can work properly until raw mode settings are saved/loaded from the scene.
+	//_html_entered(c_text->get_text());
+	//_focus_exit();
```
A proper fix is better than a temporary work-around. Perhaps `if (!raw_mode_enabled) {` should be used here too?

chaosdrop replied:
It can't be at the moment as the raw mode is not saved permanently so as soon as you switch nodes it doesn't work. I'd add the ability to permanently save raw mode status but that is beyond my abilities. As no one else who could fix it better has shown interest nor has their been a reply from the person who caused the issue to began with I figured I'd at lest fix it for the moment as it's quite an annoying issue.

aaronfranke reviewed (changes requested):
Please make the title of your PR more descriptive (perhaps "Fix Color Picker corrupting RAW mode colors #21860"). By the way, I think you meant to say "Interim".

Commit:
Fix Color Picker corrupting RAW mode colors #21860

Fix for issue #21860. So that raw mode can work properly until raw mode settings are saved/loaded from the scene.

chaosdrop commented:
Bump, made the requested changes, but think I did it the wrong way since the system still says they are their but when I click on anything related to them it just shows. "We went looking everywhere, but couldn’t find those commits."

volzhs commented:
updated OP to referer original issue

akien-mga commented:
Superseded by #25355.
